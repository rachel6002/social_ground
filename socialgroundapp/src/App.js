import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Profile from './components/profile';
import Upload from './components/upload';
import Login from './components/login';
import Registration from './components/registration';
import SuccessfulRegistration from './components/successfulRegistration';
// import UpdatePassword from './components/updatePassword';
// import PasswordReset from './components/passwordReset';
import Feed from './components/ActivityFeed';
import Find from './components/FindFriends';
import NotFound from './components/NotFound';
import PostEdit from './components/PostEdit';
import Setting from './components/Setting';
// import CaptionEdit from './components/CaptionEdit';
import Welcome from './components/Welcome';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route exact path="/" element={<Login />} />
        <Route path="/registration" element={<Registration />} />
        <Route path="/successful" element={<SuccessfulRegistration />} />
        {/* <Route path="/update" element={<UpdatePassword />} />
        <Route path="/reset" element={<PasswordReset />} /> */}
        <Route path="/profile" element={<Profile />} />
        <Route path="/post/:postId" element={<PostEdit />} />
        <Route path="/upload" element={<Upload />} />
        <Route path="/Feed" element={<Feed />} />
        <Route path="/Find" element={<Find />} />
        <Route path="/setting" element={<Setting />} />
        {/* <Route path="/caption" element={<CaptionEdit />} /> */}
        <Route path="/Welcome" element={<Welcome />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  );
}

export default App;
