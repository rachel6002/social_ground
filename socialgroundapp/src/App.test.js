/**
* @jest-environment jsdom
*/
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import App from './App';

test('SocialGround home page matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <App />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
