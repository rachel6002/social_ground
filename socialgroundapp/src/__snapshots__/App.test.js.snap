// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialGround home page matches snapshot 1`] = `
<div
  className="App"
>
  <div
    id="react-root"
  >
    <div
      className="login"
    >
      <div
        className="form-container"
      >
        <form
          className="login-form"
          id="login-form"
          method="post"
          onSubmit={[Function]}
        >
          <div
            className="large-sg-header-homepage-redirect"
          >
            <a
              href="/"
            >
              <div
                aria-disabled="false"
              >
                <img
                  alt="large_logo"
                  className="large-logo"
                  src={Object {}}
                />
              </div>
            </a>
          </div>
          <div
            className="errorMessage"
          />
          <div
            className="form-divider"
          >
            <div
              className="form-row-container"
            >
              <div
                className="form-cell"
              >
                <label
                  className="form-entry-box"
                  htmlFor="formInput1"
                >
                  <span
                    className="form-fields"
                  >
                    Enter Your Email
                  </span>
                  <input
                    aria-label="Enter Your Email"
                    aria-required="true"
                    autoCapitalize="off"
                    autoCorrect="off"
                    className="form-field-entry"
                    data-testid="loginUserEmail"
                    id="formInput1"
                    maxLength="100"
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </label>
              </div>
            </div>
            <div
              className="form-row-container"
            >
              <div
                className="form-cell"
              >
                <label
                  className="form-entry-box"
                  htmlFor="formInput2"
                >
                  <span
                    className="form-fields"
                  >
                    Enter Your Password
                  </span>
                  <input
                    aria-label="Enter Your Password"
                    aria-required="true"
                    autoCapitalize="off"
                    autoCorrect="off"
                    className="form-field-entry"
                    data-testid="loginUserPw"
                    id="formInput2"
                    maxLength="100"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </label>
              </div>
            </div>
            <br />
            <div
              className="form-submit"
            >
              <button
                className="submit-button"
                type="submit"
              >
                <div
                  className="button-label"
                >
                  Log In
                </div>
              </button>
            </div>
          </div>
          <div
            className="sign-up-container"
          >
            <p
              className="sign-up-redirect"
            >
              Don't have an account?Â 
              <a
                href="./registration"
              >
                <span
                  className="sign-up"
                >
                  Sign up
                </span>
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
